### PR TITLE
Added target name to output path to support multiple builds of same module

### PR DESCRIPTION
**Pull request by tymurmustafaiev**
**Why it's needed?**
To be able to make builds of same framework with different module names, we need to modify output path so bazel won't have conflicts.
Right now, if we call:
`bazelisk build //PathToModule/ModuleName:Target1 //PathToModule/ModuleName:Target2`
where `Target1` and `Target2` are targets with different files but same module name
bazel will fail because it will try to build 2 different frameworks to same output.
By adding `--features=swift.add_target_name_to_output`, the output will be generated to separate folders, and as output, we will have 2 different framework binaries with different files, but same framework name

### CODE REVIEW

File: swift/internal/debugging.bzl [hunk 1]
```diff
@@ -58,12 +59,18 @@ def ensure_swiftmodule_is_embedded(
...
+            feature_name = SWIFT_FEATURE_ADD_TARGET_NAME_TO_OUTPUT,
+        )
+
         # For ELF-format binaries, we need to invoke a Swift modulewrap action
         # to wrap the .swiftmodule file in a .o file that gets propagated to the
         # linker.
         modulewrap_obj = derived_files.modulewrap_object(
             actions,
             target_name = label.name,
+            add_target_name_to_output_path = add_target_name_to_output_path,
```
Comment (brentleyjones) [inline swift/internal/debugging.bzl:73]: Can you move this to be alphabetical?

File: swift/internal/derived_files.bzl [hunk 2]
```diff
@@ -17,84 +17,132 @@
...
-    return actions.declare_file("{}.autolink".format(target_name))
+    return _declare_file(
+        actions,
+        target_name,
+        "{}.autolink".format(target_name),
+        add_target_name_to_output_path,
+    )
 
-def _executable(actions, target_name):
+def _executable(actions, target_name, add_target_name_to_output_path):
```
Comment (brentleyjones) [inline swift/internal/derived_files.bzl:74]: Can you move the new parameter to be alphabetical? Same for the rest of the functions that were changed?

**Review by brentleyjones — COMMENTED** [review]
I'm fine with this since it's behind a flag, and I understand the desire.

Thoughts @keith?

File: swift/internal/derived_files.bzl (hunk 2 above)
Comment (tymurmustafaiev) [reply swift/internal/derived_files.bzl]: @brentleyjones done, thx!

File: swift/internal/debugging.bzl (hunk 1 above)
Comment (tymurmustafaiev) [reply swift/internal/debugging.bzl:72]: @brentleyjones done, thx!

**Comment by keith** [comment]
CI failures look real

**Comment by tymurmustafaiev** [comment]
@keith @brentleyjones 👋 hey guys, thanks for review, I've fixed tests
Was related to latest changes in master so I had to update them as well 🙏

**Comment by keith** [comment]
LGTM, @brentleyjones can merge unless there are more comments

File: swift/internal/compiling.bzl [hunk 3]
```diff
@@ -2990,7 +3030,8 @@ def _declare_multiple_outputs_and_write_output_file_map(
         emits_bc,
         split_derived_file_generation,
         srcs,
-        target_name):
+        target_name,
+        add_target_name_to_output_path):
```
Comment (brentleyjones) [inline swift/internal/compiling.bzl:3034]: Can you move this up to before `emits_bc` to keep it alphabetical? And then adjust the call sites as well?

**Review by brentleyjones — APPROVED** [review]
Just one more formatting thing I want, then g2g.

File: swift/internal/compiling.bzl (hunk 3 above)
Comment (tymurmustafaiev) [reply swift/internal/compiling.bzl]: @brentleyjones sure, done 🙏

**Comment by keith** [comment]
Found a regression with this detailed in the description here: https://github.com/bazelbuild/rules_swift/pull/1097

you can fixup and submit a revert of that PR whenever you're ready!

**Pull request merged**